frontend/src/components/messages/AssistanceRequestTable.vue

```diff
@@ -0,0 +1,193 @@
+    }
+  },
+  computed: {
+    ...mapState(useMessagesStore, ['assistanceRequests']),
+  },
+  watch: {
+    markReadButtonState: {
+      handler() {
+        this.updateBodyCheckboxesReadUnread(true)
+      },
+    },
+    markUnreadButtonInMessageTableState: {
+      handler() {
+        this.updateBodyCheckboxesReadUnread(false)
+      },
+    },
+    markUnreadButtonInConversationThreadState: {
+      async handler() {
+        await this.updateMessageReadUnread(false, this.selectedRequestId)
+      },
+    },
+  },
+  methods: {
+    ...mapActions(useMessagesStore, ['updateAssistanceRequest']),
+
+    resetAllCheckboxes() {
+      this.bodyCheckboxesSelected = []
+    },
+    /**
+     * Update the body/item checkboxes to read or unread.
+     */
+    async updateBodyCheckboxesReadUnread(isRead) {
+      const selectedAssistanceRequestIds = this.bodyCheckboxesSelected
+      this.resetAllCheckboxes()
+      await Promise.all(
+        selectedAssistanceRequestIds.map(async (assistanceRequestId) => {
+          await this.updateMessageReadUnread(isRead, assistanceRequestId)
+        }),
+      )
+    },
+    /**
+     * Handles the row click event. When a row is clicked the assistanceRequests in context is displayed and marked as read.
+     */
+    async rowClickHandler(item, row) {
+      this.selectedRequestId = row?.item?.assistanceRequestId
+      this.$emit('openRequestConversation', this.selectedRequestId)
+      await this.updateMessageReadUnread(true, this.selectedRequestId)
+    },
+    async updateMessageReadUnread(isRead, assistanceRequestId) {
+      this.selectedAssistanceRequest = this.assistanceRequests?.find((item) => item.assistanceRequestId === assistanceRequestId)
+      let payload = {}
+      if (isRead) {
+        this.selectedAssistanceRequest.lastOpenedTime = new Date().toUTCString()
+        payload.lastOpenedTime = this.selectedAssistanceRequest.lastOpenedTime
+      }
+      if (this.selectedAssistanceRequest.isRead != isRead) {
+        this.selectedAssistanceRequest.isRead = isRead
+        payload.isRead = isRead
+      }
+      await this.updateAssistanceRequest(assistanceRequestId, payload)
+    },
+    isActionRequiredMessage(item) {
+      return item?.status === 'Action required'
+    },
+    getItemClass(item) {
+      let result = ''
+      if (this.isActionRequiredMessage(item)) {
+        result += 'action-required-message '
+      }
+      if (!item?.isRead) {
+        result += 'unread-message '
+      }
+      if (this.selectedRequestId === item?.assistanceRequestId) {
+        result += 'highlighted-row '
+      }
+      return result
+    },
+    convertDateToLocalDate(date) {
+      let localDate = new Date(date)
+      let monthNames = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
+      let dateString = localDate?.getFullYear() + '-' + monthNames[localDate?.getMonth()] + '-' + localDate?.getDate()
+      return dateString
+    },
```

Comment:
Can we bring in moment() or another library to do date formatting/parsing? Also this might be best in utils if there's a lot of code. Though if it's just a simple in moment then we can leave it here.